packages/api/src/routers/external-api/__tests__/charts.test.ts

```diff
@@ -0,0 +1,821 @@
+    });
+
+    it('should return 400 if series array exceeds max length (5)', async () => {
+      const payload = createSeriesRequestPayload(logSource.id.toString(), {
+        series: Array(6).fill({ aggFn: 'count', where: '', groupBy: [] }),
+      });
+      await authRequest('post', BASE_URL).send(payload).expect(400);
+    });
+
+    it('should return 400 if groupBy fields are inconsistent across series', async () => {
+      const payload = createSeriesRequestPayload(logSource.id.toString(), {
+        series: [
+          { aggFn: 'count', where: '', groupBy: ['field1'] },
+          { aggFn: 'count', where: '', groupBy: ['field2'] },
+        ],
+      });
+      await authRequest('post', BASE_URL).send(payload).expect(400);
+    });
+
+    it('should return 400 for invalid aggFn', async () => {
+      const payload = createSeriesRequestPayload(logSource.id.toString(), {
+        series: [{ aggFn: 'invalid_function', where: '', groupBy: [] }],
+      });
+      await authRequest('post', BASE_URL).send(payload).expect(400);
+    });
+
+    // =======================================
+    // Data Querying Tests (Logs as Source)
+    // =======================================
+    it('should return total count when where clause is empty', async () => {
+      // Note: counts only info logs (2) + metric logs (2) = 4 total
+      const payload = createSeriesRequestPayload(logSource.id.toString());
+      const response = await authRequest('post', BASE_URL)
+        .send(payload)
+        .expect(200);
+
+      expect(response.body.data).toBeInstanceOf(Array);
+      expect(response.body.data).toHaveLength(1);
+      expect(Number(response.body.data[0]['series_0.data'])).toEqual(4);
+    });
+
+    it('should handle lucene where clause (default)', async () => {
+      const payload = createSeriesRequestPayload(logSource.id.toString(), {
+        series: [
+          {
+            aggFn: 'count',
+            where: 'SeverityText:info AND ServiceName:"test-service-2"',
+            groupBy: [],
+          },
+        ],
+      });
+      const response = await authRequest('post', BASE_URL)
+        .send(payload)
+        .expect(200);
+      expect(response.body.data).toHaveLength(1);
+      expect(Number(response.body.data[0]['series_0.data'])).toEqual(1);
+    });
+
+    it('should handle sql where clause', async () => {
+      const payload = createSeriesRequestPayload(logSource.id.toString(), {
+        series: [
+          {
+            aggFn: 'count',
+            where: "SeverityText = 'info' AND Body LIKE '%message 2%'",
+            whereLanguage: 'sql',
+            groupBy: [],
+          },
+        ],
+      });
+      const response = await authRequest('post', BASE_URL)
+        .send(payload)
+        .expect(200);
+      expect(response.body.data).toHaveLength(1);
+      expect(Number(response.body.data[0]['series_0.data'])).toEqual(1);
+    });
+
+    it('should handle lucene query errors gracefully', async () => {
```

Comment:
Should there be a corresponding test to handle sql errors - e.g. what does it return when I give it a non-sql where clause when `whereLanguage == 'sql'`?